tensorflow_addons/layers/multihead_attention.py

```diff
@@ -0,0 +1,253 @@
+        bias_regularizer: typing.Union[str, typing.Callable] = None,
+        bias_constraint: typing.Union[str, typing.Callable] = None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.head_size = head_size
+        self.num_heads = num_heads
+        self.output_size = output_size
+        self.dropout_rate = dropout_rate
+        self.use_projection_bias = use_projection_bias
+        self.return_attn_coef = return_attn_coef
+
+        self.kernel_initializer = tf.keras.initializers.get(kernel_initializer)
+        self.kernel_regularizer = tf.keras.regularizers.get(kernel_regularizer)
+        self.kernel_constraint = tf.keras.constraints.get(kernel_constraint)
+        self.bias_initializer = tf.keras.initializers.get(bias_initializer)
+        self.bias_regularizer = tf.keras.regularizers.get(bias_regularizer)
+        self.bias_constraint = tf.keras.constraints.get(bias_constraint)
+
+    def build(self, input_shape):
+
+        num_query_features = input_shape[0][-1]
+        num_key_features = input_shape[1][-1]
+        num_value_features = input_shape[2][-1]
+        output_size = self.output_size if self.output_size else num_value_features
+
+        self.query_kernel = self.add_weight(
+            name="query_kernel",
+            shape=[self.num_heads, num_query_features, self.head_size],
+            initializer=self.kernel_initializer,
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+        )
+        self.key_kernel = self.add_weight(
+            name="key_kernel",
+            shape=[self.num_heads, num_key_features, self.head_size],
+            initializer=self.kernel_initializer,
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+        )
+        self.value_kernel = self.add_weight(
+            name="value_kernel",
+            shape=[self.num_heads, num_value_features, self.head_size],
+            initializer=self.kernel_initializer,
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+        )
+        self.projection_kernel = self.add_weight(
+            name="projection_kernel",
+            shape=[self.num_heads, self.head_size, output_size],
+            initializer=self.kernel_initializer,
+            regularizer=self.kernel_regularizer,
+            constraint=self.kernel_constraint,
+        )
+
+        if self.use_projection_bias:
+            self.projection_bias = self.add_weight(
+                name="projection_bias",
+                shape=[output_size],
+                initializer=self.bias_initializer,
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+            )
+        else:
+            self.projection_bias = None
+
+        self.dropout = tf.keras.layers.Dropout(self.dropout_rate)
+
+        super().build(input_shape)
+
+    def call(self, inputs, training=None):
```

Comment:
There might should be param for masking as well, if you layer support input masking.

Reply:
Ah yes, its named [mask](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#__call__)